### PR TITLE
Feature/recommendation endpoint

### PR DESCRIPTION
**Pull request by gorzala**

### CODE REVIEW

File: src/main/java/net/dancier/domain/Recommendable.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/Recommendable.java:3]: Add doc, please

File: src/main/java/net/dancier/domain/Recommendation.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/Recommendation.java:6]: Add doc, please

File: src/main/java/net/dancier/domain/Recommendation.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/Recommendation.java:19]: I suggest extending the enum with one or more class references for auto resolving instead of manual mappping.

Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/Recommendation.java:19]: Add doc please. Document or better use use annotation to inform about null handling. Right now null will lead to ETC.

File: src/main/java/net/dancier/domain/Recommendation.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/Recommendation.java:13]: What is this? Should be documented. Is this just "everything that is not defined yet". Should this be handled at all? (Personally I would throw an exception in such a case I think)

File: src/main/java/net/dancier/domain/dance/School.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/dance/School.java:12]: Doc, please

File: src/main/java/net/dancier/domain/dance/School.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/dance/School.java:15]: You should document (best with NotNull, NotrEmpty or NonNull annoations) what is

File: src/main/java/net/dancier/domain/dance/School.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/domain/dance/School.java:13]: Documentr the meaning of this user, please. What' the purpose of this field?

File: src/main/java/net/dancier/resources/RecommendationsResource.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/resources/RecommendationsResource.java:26]: One could argue if this is a good name ;) I'd suggest something like readRecommendation or getRecommandation or getSingleRecord or something.

File: src/main/java/net/dancier/service/MockedRecommendationsService.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/service/MockedRecommendationsService.java:11]: A short doc about mocked behaviour here would be nice.

File: src/main/java/net/dancier/service/RecommendationsService.java
Comment (DanielVogtland) [inline src/main/java/net/dancier/service/RecommendationsService.java:8]: Doc, please.

File: src/test/java/net/dancier/MainTest.java
Comment (DanielVogtland) [inline src/test/java/net/dancier/MainTest.java:30]: Why not using Junit assert?

**Review by DanielVogtland — COMMENTED** [review]
Please use documentation. And I would suggest more use of describing annotations.

File: src/main/java/net/dancier/domain/Recommendable.java
Comment (gorzala) [reply src/main/java/net/dancier/domain/Recommendable.java:3]: Done

File: src/main/java/net/dancier/domain/Recommendation.java
Comment (gorzala) [reply src/main/java/net/dancier/domain/Recommendation.java:13]: I decided to follow your personal preference by throwing an exeption.

File: src/main/java/net/dancier/domain/dance/School.java
Comment (gorzala) [reply src/main/java/net/dancier/domain/dance/School.java:12]: OK, I am getting to know you ;-)

File: src/main/java/net/dancier/domain/dance/School.java
Comment (gorzala) [reply src/main/java/net/dancier/domain/dance/School.java:13]: I try to have my code selfdescribing ;-) (Not that I claim it is ;-) )
I renamed the field to owner, is this OK for you?

File: src/main/java/net/dancier/domain/dance/School.java
Comment (gorzala) [reply src/main/java/net/dancier/domain/dance/School.java:15]: Did not get what you want, maybe we have a chat tomorrow or whatever.

File: src/main/java/net/dancier/resources/RecommendationsResource.java
Comment (gorzala) [reply src/main/java/net/dancier/resources/RecommendationsResource.java:26]: would you be happy with: readRecommandations?

File: src/main/java/net/dancier/service/MockedRecommendationsService.java
Comment (gorzala) [reply src/main/java/net/dancier/service/MockedRecommendationsService.java:11]: yes

File: src/main/java/net/dancier/service/RecommendationsService.java
Comment (gorzala) [reply src/main/java/net/dancier/service/RecommendationsService.java:8]: done

File: src/test/java/net/dancier/MainTest.java
Comment (gorzala) [reply src/test/java/net/dancier/MainTest.java:30]: I just used it here honestly, because I followed an example for writing an Int-Test for dropwizard.
In general I have no strong feelings about what to use, although I slightly thing that assertj statements are a little bit more readable, and also testing for collections is (I think) someway more easy.
If you have more strong feelings, just tell: I will follow you ;-)

File: src/main/java/net/dancier/domain/Recommendation.java
Comment (gorzala) [reply src/main/java/net/dancier/domain/Recommendation.java:19]: I a currently wondering, if we need this wrapper at all. So give me some time. When we will stick to this solution with recommendation, I will consider your suggestion.

Comment (gorzala) [reply src/main/java/net/dancier/domain/Recommendation.java:22]: Ok, I think we still need this wrapper.
What do you mean with the adding a class reference to the enum:
Dancer("dancer", Dancer.class) ?
How do I get autoresolving then?

File: src/test/java/net/dancier/MainTest.java
Comment (Qwby) [reply src/test/java/net/dancier/MainTest.java:30]: I would always use AssertJ over JUnit asserts cause they read fluently. And they have a lot of nice helper functions :)

**Review by Qwby — APPROVED** [review]

File: src/main/java/net/dancier/domain/dance/School.java
Comment (DanielVogtland) [reply src/main/java/net/dancier/domain/dance/School.java:16]: Self describing code usually is an illusion ;-)

File: src/test/java/net/dancier/MainTest.java
Comment (DanielVogtland) [reply src/test/java/net/dancier/MainTest.java:30]: I prefer JUnit + Hamcrest (fluent enough for my taste and nice helpers) but not knowing the power of AssertJ. Just very happy with that combination and would be my choice - but would use AssertJ, too.

Comment (gorzala) [reply src/test/java/net/dancier/MainTest.java:30]: So in the end consent of all for AssertJ. Fine ;-)

File: src/main/java/net/dancier/resources/RecommendationsResource.java
Comment (DanielVogtland) [reply src/main/java/net/dancier/resources/RecommendationsResource.java]: Absolutely

File: src/main/java/net/dancier/domain/dance/School.java
Comment (DanielVogtland) [reply src/main/java/net/dancier/domain/dance/School.java:16]: By the way: owner would be great

File: src/main/java/net/dancier/domain/dance/School.java
Comment (gorzala) [reply src/main/java/net/dancier/domain/dance/School.java:18]: Did it

File: src/main/java/net/dancier/resources/login/LoginResource.java
Comment (gorzala) [inline src/main/java/net/dancier/resources/login/LoginResource.java:142]: Without this I had the situation, that wen devEnv is enabled, any real oauth-redirect stopped working completly

File: src/main/java/net/dancier/service/RecommendationsService.java
Comment (gorzala) [reply src/main/java/net/dancier/service/RecommendationsService.java:9]: OK, with the documentation on a method level? Could not think of anything really useful an the interface Level.

Comment (DanielVogtland) [reply src/main/java/net/dancier/service/RecommendationsService.java:9]: I really would write something like "Service for accessing recommendations." but I know you would reject because no real new info ;-) It's kind of an esthetic issue for me at least. But although I would and will write such a comment I won't force you. :-)

**Review by DanielVogtland — APPROVED** [review]

File: src/main/java/net/dancier/service/RecommendationsService.java
Comment (gorzala) [reply src/main/java/net/dancier/service/RecommendationsService.java:12]: Added this kind of documentation ;-)
Anything left before you can approve ;-)

File: src/main/java/net/dancier/resources/login/LoginResource.java
Comment (DanielVogtland) [reply src/main/java/net/dancier/resources/login/LoginResource.java:142]: I already approved 11 days ago ;-) But I can once again after the new commit of course.

**Review by DanielVogtland — APPROVED** [review]

**Pull request merged**
